test/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests/MacroTests/GuidMacroTests.cs

```diff
@@ -78,5 +78,39 @@ private static void ValidateGuidMacroCreatedParametersWithResolvedValues(string
                 Assert.Equal(paramValue, testValue);
             }
         }
+
+        [Fact]
+        public void TestDefaultFormatIsCaseSensetive()
+        {
+            string paramNameLower = "TestGuidLower";
+            IMacroConfig macroConfigLower = new GuidMacroConfig(paramNameLower, "string", string.Empty, "d");
+            string paramNameUpper = "TestGuidUPPER";
+            IMacroConfig macroConfigUpper = new GuidMacroConfig(paramNameUpper, "string", string.Empty, "D");
+
+            IVariableCollection variables = new VariableCollection();
+            IRunnableProjectConfig config = new SimpleConfigModel(_engineEnvironmentSettings.Host.LoggerFactory);
+            IParameterSet parameters = new ParameterSet(config);
+            ParameterSetter setter = MacroTestHelpers.TestParameterSetter(_engineEnvironmentSettings, parameters);
+
+            GuidMacro guidMacro = new GuidMacro();
+            guidMacro.EvaluateConfig(_engineEnvironmentSettings, variables, macroConfigLower, parameters, setter);
+            guidMacro.EvaluateConfig(_engineEnvironmentSettings, variables, macroConfigUpper, parameters, setter);
+
+            // Logic of below might be a bit confusing...
+            // We are just making sure that no UPPER case char is present in lower case and vice-versa
+            // Reason is... In theory GUID could be all numbers, hence can't check if UPPER char is present in UPPER guid...
```

Comment:
We can do `Assert.True( !char.IsLetter(c) || char.IsLower(c) )` and remove these comments :)

Reply:
What you wrote means it always pass I dont follow...

Reply:
It doesn't always pass. 
1. If the character is not a letter, you skip evaluation. Because digits don't have upper/lower casing.
2. If it is a character, you make sure it is lowercase. Because we want our GUID to be formatted in lowercase.

So if the character is 'C', it fails since:
1. C is a letter. The first expression evaluates to `false`
2. It is not lowercase. The second expression evaluates to `false`